src/ophyd_async/plan_stubs/ensure_connected.py

```diff
@@ -10,13 +13,18 @@ def ensure_connected(
     timeout: float = DEFAULT_TIMEOUT,
     force_reconnect=False,
 ):
-    yield from bps.wait_for(
+    connect_task_singleton: List[asyncio.Task] = yield from bps.wait_for(
         [
             lambda: wait_for_connection(
                 **{
-                    device.name: device.connect(mock, timeout, force_reconnect)
+                    device.name: device.connect(
+                        mock=mock, timeout=timeout, force_reconnect=force_reconnect
+                    )
                     for device in devices
                 }
             )
         ]
     )
+
+    if connect_task_singleton and connect_task_singleton[0].exception() is not None:
+        raise connect_task_singleton[0].exception()
```

Comment:
Wondering why you check the exception and raise rather than just
```suggestion
    connect_task.result()
```

Reply:
I was doing this before, but decided what I have now makes for nicer reading. Can change it back.

Reply:
I think the `result()` will give a slightly nicer stack trace, but not sure...